api/grpcserver/mesh_service.go

```diff
@@ -123,22 +123,21 @@ func (s MeshService) getFilteredTransactions(startLayer types.LayerID, addr type
 }
 
 func (s MeshService) getFilteredActivations(startLayer types.LayerID, addr types.Address) (activations []*types.ActivationTx, err error) {
-	// We have no way to look up activations by coinbase so we have no choice
-	// but to read all of them.
-	// TODO: index activations by layer (and maybe by coinbase)
-	// See https://github.com/spacemeshos/go-spacemesh/issues/2064.
-	var atxids []types.ATXID
-	for l := startLayer; l <= s.Mesh.LatestLayer(); l++ {
-		layer, err := s.Mesh.GetLayer(l)
-		if layer == nil || err != nil {
-			return nil, status.Errorf(codes.Internal, "error retrieving layer data")
-		}
 
-		for _, b := range layer.Blocks() {
-			if b.ActiveSet != nil {
-				atxids = append(atxids, *b.ActiveSet...)
-			}
+	var atxids []types.ATXID
+	atxIter := s.Mesh.GetAtxIterByCoinbase(addr)
```

Comment:
Iterators are a pretty low-level abstraction and I'd prefer to keep them out of the API code if possible. I think it's better if this code lives inside the Mesh, and you just call a simple function here that returns all of the ATX data (like the existing `Mesh.GetATXs()`) - something like `Mesh.GetATXsByCoinbase()`.